### PR TITLE
Add support for PCA9557 8-bit IO expander

### PR DESCRIPTION
**Pull request by cctweaker**
## Description:
This change will add a driver for the PCA9557 8-bit I2C IO expander to be used as virtual relay.
Since the PCA9557 has no interrupt pin, button and switches are not implemented.
Based on the new released MCP23017 DRV as virtual switch/button/relay.

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [ ] The code change is tested and works with Tasmota core ESP32 V.2.0.9
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

**Comment by Jason2866** [comment]
There are already enough supported IO expanders. Why add another?
Because the IC exists is not reason enough to add to Tasmota.
Every new driver adds additional maintain work.

File: tasmota/tasmota_xdrv_driver/xdrv_69_pca9557.ino [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+
+/*********************************************************************************************\
+ * PCA9557 - I2C
+\*********************************************************************************************/
+
+void PCA9557DumpRegs(void) {
+  uint8_t data[4];
+  for (Pca9557.chip = 0; Pca9557.chip < Pca9557.max_devices; Pca9557.chip++) {
+    uint32_t data_size = sizeof(data);
+#ifdef USE_I2C
```
Comment (arendst) [inline tasmota/tasmota_xdrv_driver/xdrv_69_pca9557.ino:96]: Remove these ifdefs. Already done at start of file.

**Comment by arendst** [comment]
Add buttons and switches too. Works fine without interrupts.

Report succesful functionality with your hardware before I consider merging.

**Comment by cctweaker** [comment]
> There are already enough supported IO expanders. Why add another? Because the IC exists is not reason enough to add to Tasmota. Every new driver adds additional maintain work.

You are absolutely right, there are enough but sadly I have about a dozen installed light controllers that use MCP23017 and this PCA9557 (cheaper and was available compared to MCP23008).
I think it would be easier for me to maintain this driver than to maintain a complete firmware for them.

File: tasmota/tasmota_xdrv_driver/xdrv_69_pca9557.ino [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+
+  bool result = false;
+
+  if (FUNC_SETUP_RING2 == function) {
+    PCA9557ModuleInit();
+  } else if (Pca9557.max_devices) {
+    switch (function) {
+      case FUNC_LOOP:
+      case FUNC_SLEEP_LOOP:
+      case FUNC_EVERY_100_MSECOND:
```
[the remark this answers is not in the document]
Comment (cctweaker) [reply tasmota/tasmota_xdrv_driver/xdrv_69_pca9557.ino:513]: Done, thank you!

**Pull request merged**